agents/exec/installer/src/main/resources/installers/1.0.1/org.eclipse.che.exec.script.sh

```diff
@@ -0,0 +1,224 @@
+    >&2 uname -a
+    exit 1
+fi
+
+# Compute URI of workspace master
+WORKSPACE_MASTER_URI=$(echo $CHE_API | cut -d / -f 1-3)
+
+## Evaluate variables now that prefix is defined
+eval "LOCAL_AGENT_BINARIES_URI=${LOCAL_AGENT_BINARIES_URI}"
+eval "DOWNLOAD_AGENT_BINARIES_URI=${DOWNLOAD_AGENT_BINARIES_URI}"
+eval "TARGET_AGENT_BINARIES_URI=${TARGET_AGENT_BINARIES_URI}"
+
+LOCAL_AGENT_PATH=
+if [ -f "${LOCAL_AGENT_BINARIES_URI}" ]; then
+    AGENT_BINARIES_URI="file://${LOCAL_AGENT_BINARIES_URI}"
+    LOCAL_AGENT_PATH=${LOCAL_AGENT_BINARIES_URI}
+elif [ -f $(echo "${LOCAL_AGENT_BINARIES_URI}" | sed "s/-${PREFIX}//g") ]; then
+    AGENT_BINARIES_URI="file://"$(echo "${LOCAL_AGENT_BINARIES_URI}" | sed "s/-${PREFIX}//g")
+    LOCAL_AGENT_PATH=$(echo "${LOCAL_AGENT_BINARIES_URI}" | sed "s/-${PREFIX}//g")
+else
+    AGENT_BINARIES_URI=${DOWNLOAD_AGENT_BINARIES_URI}
+fi
+
+# If file is already on the filesystem, use it
+if [ ! -z ${LOCAL_AGENT_PATH} ]; then
+  tar zxf ${LOCAL_AGENT_PATH} -C ${CHE_DIR}
+else
+  echo "Exec Agent binary is downloaded remotely"
+  # Use curl
+  if [ ${CURL_INSTALLED} = true ]; then
+    if curl -o /dev/null --silent --head --fail $(echo ${AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g'); then
+      curl -o $(echo ${TARGET_AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g' | sed 's/file:\/\///g') -s $(echo ${AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g')
+    elif curl -o /dev/null --silent --head --fail $(echo ${AGENT_BINARIES_URI} | sed 's/-\${PREFIX}//g'); then
+      curl -o $(echo ${TARGET_AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g' | sed 's/file:\/\///g') -s $(echo ${AGENT_BINARIES_URI} | sed 's/-\${PREFIX}//g')
+    fi
+    curl -s $(echo ${TARGET_AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g') | tar  xzf - -C ${CHE_DIR}
+  else
+    # replace https by http as wget may not be able to handle ssl
+    AGENT_BINARIES_URI=$(echo ${AGENT_BINARIES_URI} | sed 's/https/http/g')
+
+    # use wget
+    WGET_SPIDER="wget --spider"
+    if wget  2>&1 | grep -q BusyBox; then
+      WGET_SPIDER="wget -s"
+    fi
+    LOCAL_DOWNLOAD=$(echo ${TARGET_AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g' | sed 's/file:\/\///g')
+    if ${WGET_SPIDER} -q $(echo ${AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g') >/dev/null; then
+      wget -qO ${LOCAL_DOWNLOAD} $(echo ${AGENT_BINARIES_URI} | sed 's/\${PREFIX}/'${PREFIX}'/g')
+    elif ${WGET_SPIDER} -q $(echo ${AGENT_BINARIES_URI} | sed 's/-\${PREFIX}//g'); then
+      wget -qO- ${LOCAL_DOWNLOAD} $(echo ${AGENT_BINARIES_URI} | sed 's/-\${PREFIX}//g')
+    fi
+    tar xzf ${LOCAL_DOWNLOAD} -C ${CHE_DIR}
+  fi
+fi
+
+
+
+if [ -f /bin/bash ]; then
+    SHELL_INTERPRETER="/bin/bash"
+fi
+
+EXEC_AGENT_PORT=${CHE_SERVER_EXEC_AGENT_HTTP_PORT:-4412}
+
+#####################################################
+### exec-agent run command will be added here ###
+#####################################################
```

Comment:
Could you pls point out what has been modified/added here? Since most of all it is copy-paste.

Reply:
It is copied/pasted from the previous version of the script. As far as I know, the script is not modified anymore. Good to time to remove this comment. I'll do it.